spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/web/TraceFilterTests.java

```diff
@@ -382,6 +382,38 @@ public void returns200IfSpanParentIsMalformedAndCreatesANewSpan() throws Excepti
 		then(this.response.getStatus()).isEqualTo(HttpStatus.OK.value());
 	}
 
+	@Test
+	public void samplesASpanRegardlessOfTheSamplerWhenXB3FlagsIsPresentAndSetTo1() throws Exception {
+		this.request = builder()
+				.header(Span.SPAN_FLAGS, 1)
+				.buildRequest(new MockServletContext());
+		this.sampler = new NeverSampler();
+		TraceFilter filter = new TraceFilter(this.tracer, this.traceKeys, this.spanReporter,
+				this.spanExtractor, this.httpTraceKeysInjector);
+
+		filter.doFilter(this.request, this.response, this.filterChain);
+
+		then(new ListOfSpans(this.spanReporter.getSpans())).allSpansAreExportable();
+		then(TestSpanContextHolder.getCurrentSpan()).isNull();
+		then(ExceptionUtils.getLastException()).isNull();
+	}
+
+	@Test
+	public void doesNotOverrideTheSampledFlagWhenXB3FlagIsSetToOtherValueThan1() throws Exception {
+		this.request = builder()
+				.header(Span.SPAN_FLAGS, 0)
+				.buildRequest(new MockServletContext());
+		this.sampler = new AlwaysSampler();
+		TraceFilter filter = new TraceFilter(this.tracer, this.traceKeys, this.spanReporter,
+				this.spanExtractor, this.httpTraceKeysInjector);
+
+		filter.doFilter(this.request, this.response, this.filterChain);
+
+		then(new ListOfSpans(this.spanReporter.getSpans())).allSpansAreExportable();
+		then(TestSpanContextHolder.getCurrentSpan()).isNull();
+		then(ExceptionUtils.getLastException()).isNull();
+	}
+
```

Comment:
ah so you're saying that we need to update the collectors to check for the presence of that flag?